### PR TITLE
[WIP]Make TestFx repo CPS Based

### PR DESCRIPTION
**Pull request by mayankbansal018**
Things to verify before pushing this:

- Localization
- Assembly, & File Version
- Nugets Upgradation
- Vsix Templates / Wizard 
- Run on VSO

### CODE REVIEW

**Comment by mayankbansal018** [comment]
@dotnet-bot test this please

File: Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <TestFxRoot Condition="$(TestFxRoot) == ''">..\..\..\</TestFxRoot>
   </PropertyGroup>
-  <Import Project="$(TestFxRoot)scripts\build\TestFx.Settings.targets" />
+  <Import Project="$(TestFxRoot)scripts\build\TestFx.Settings.Templates.targets" />
```
Comment (jayaranigarg) [inline Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj:6]: Why did we separate them?

File: scripts/toolset/packages.config [hunk 2]
```diff
@@ -6,4 +6,6 @@
   <package id="MSTest.Internal.TestFx.Localized.Documentation" version="1.0.0-build-20170420-1" />
   <package id="vswhere" version="2.0.2" />
   <package id="Pdb2Pdb" version="1.1.0-beta1-62316-01" />
+  <package id="Microsoft.TestPlatform" version="15.5.0" />
```
Comment (jayaranigarg) [inline scripts/toolset/packages.config:9]: Shouldn't we use 15.7 instead?

File: src/Adapter/PlatformServices.Interface/PlatformServices.Interface.csproj [hunk 3]
```diff
@@ -1,69 +1,28 @@
...
-      <Private>False</Private>
-    </ProjectReference>
+    <ProjectReference Include="..\..\TestFramework\Extension.Core\Extension.Core.csproj" />
+    <ProjectReference Include="..\..\TestFramework\MSTest.Core\MSTest.Core.csproj" />
   </ItemGroup>
-  <ItemGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net45' ">
+    <PackageReference Include="Microsoft.TestPlatform.ObjectModel" Version="11.0.1" />
+  </ItemGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.3' ">
```
Comment (jayaranigarg) [inline src/Adapter/PlatformServices.Interface/PlatformServices.Interface.csproj]: Why a conditional package reference here?

File: src/Adapter/PlatformServices.NetCore/PlatformServices.NetCore.csproj [hunk 4]
```diff
@@ -33,7 +32,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.TestPlatform.ObjectModel" Version="11.0.0" />
+    <PackageReference Include="Microsoft.TestPlatform.ObjectModel" Version="11.0.1" />
```
Comment (jayaranigarg) [inline src/Adapter/PlatformServices.NetCore/PlatformServices.NetCore.csproj:35]: what is the difference between 11.0 and 11.1?

File: src/Adapter/PlatformServices.Portable/PlatformServices.Portable.csproj [hunk 5]
```diff
@@ -1,43 +1,24 @@
...
-    <WarningLevel>4</WarningLevel>
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <TargetFrameworks>netstandard1.3;net45</TargetFrameworks>
```
Comment (jayaranigarg) [inline src/Adapter/PlatformServices.Portable/PlatformServices.Portable.csproj:11]: why are we building this for net45?

File: src/Package/MSTest.TestAdapter.Enu.nuspec [hunk 6]
```diff
@@ -28,21 +28,27 @@
...
-    <file src="MSTest.CoreAdapter\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.dll" target="build\_common\" />
-    <file src="PlatformServices.Interface\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.Interface.dll" target="build\_common\" />
-    <file src="PlatformServices.Desktop\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.dll" target="build\_common\" />
-    <file src="MSTest.Core\Microsoft.VisualStudio.TestPlatform.TestFramework.dll" target="build\_common\" />
 
     <!-- NetCore -->
-    <file src="Build\NetCore\MSTest.TestAdapter.props" target="build\netcoreapp1.0\" />
+    <file src="MSTest.CoreAdapter\netstandard1.3\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.dll" target="build\netcoreapp1.0\" />
+    <file src="PlatformServices.Interface\netstandard1.3\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.Interface.dll" target="build\netcoreapp1.0\" />
+    <file src="MSTest.Core\netstandard1.1\Microsoft.VisualStudio.TestPlatform.TestFramework.dll" target="build\net45\" />
```
Comment (jayaranigarg) [inline src/Package/MSTest.TestAdapter.Enu.nuspec:35]: Nit: build\netcoreapp1.0\

File: src/Package/MSTest.TestAdapter.nuspec [hunk 7]
```diff
@@ -28,64 +28,72 @@
...
-    <file src="MSTest.CoreAdapter\it\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll" target="\build\_common\it" />
-    <file src="MSTest.CoreAdapter\ja\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll" target="\build\_common\ja" />
-    <file src="MSTest.CoreAdapter\ko\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll" target="\build\_common\ko" />
-    <file src="MSTest.CoreAdapter\pl\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll" target="\build\_common\pl" />
-    <file src="MSTest.CoreAdapter\pt-BR\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll" target="\build\_common\pt" />
-    <file src="MSTest.CoreAdapter\ru\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll" target="\build\_common\ru" />
-    <file src="MSTest.CoreAdapter\tr\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll" target="\build\_common\tr" />
+    <!-- Resources -->
+    <file src="MSTest.CoreAdapter\netstandard1.3\zh-Hans\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll" target="\build\_common\zh-Hans" />
```
Comment (jayaranigarg) [inline src/Package/MSTest.TestAdapter.nuspec:33]: Will discuss on where to keep resources offline..

File: test/ComponentTests/PlatformServices.Desktop.Component.Tests/ReflectionUtilityTests.cs [hunk 8]
```diff
@@ -42,7 +42,6 @@ public ReflectionUtilityTests()
             var testAssetPath =
                 Path.Combine(
                     Directory.GetParent(Directory.GetParent(Directory.GetParent(currentAssemblyPath).FullName).FullName).FullName,
-                    "artifacts",
```
Comment (jayaranigarg) [inline test/ComponentTests/PlatformServices.Desktop.Component.Tests/ReflectionUtilityTests.cs:45]: ??

**Comment by singhsarab** [comment]
@mayankbansal018 What is the status of this PR?

**Comment by jayaranigarg** [comment]
@mayankbansal018 : When are we planning to push this PR in? This issue https://github.com/Microsoft/testfx/issues/481 will also get resolved as part of this PR.

**Comment by parrainc** [comment]
@mayankbansal018 are there any news about this PR?
Some opened issues depend on this changes in order to be fixed.

**Comment by mayankbansal018** [comment]
@parrainc , Unfortunately I've not been able to prioritize this task, due to other tasks that I've in hand. As per the last status of this PR, I had moved the project to CPS, & everything was running fine. 

I understand that there are a few bugs pending on this, because of which I would encourage if you would like to take this further, let me know if that would be possible, I would give you permission on my branch.

**Comment by parrainc** [comment]
@mayankbansal018 i understand.
You can count me in, it would be a pleasure to take this further. I'll make sure to get this done asap, so those few issues that depends on this changes got closed soon :+1:

in any case, if questions arise on the road, i'll be commenting here :)

**Comment by parrainc** [comment]
@mayankbansal018 quick question... is there a specific reason of why we are upgrading to netstandard 1.3 instead of a major version (i.e. 1.5, 1.6) or the latest one (2.0)? I'm not sure if latest or a different version than 1.3 is not fully compatible with some of the projects in the solution. If that's not the case then I'm just curious to know what could be done about it.

**Comment by dotMorten** [comment]
@parrainc why upgrade to a version higher than needed? The higher the version, the fewer the platforms can use it

**Comment by mayankbansal018** [comment]
@parrainc , as dotMorten mentioned, the lower the version for .net standard, the wider the range for platforms we can cover. For e.g. if we move to > NS1.4, then it won't work with UWP, & we would need to have separate build scripts to fix that.

**Comment by parrainc** [comment]
gotcha

**Comment by parrainc** [comment]
@mayankbansal018 quick update:
did some change to sync up master with testFxCPS branch, and everything is going fine (building ok in VS, and unit and integration tests seems to behave just fine). The only issue I'm having is that while building from CLI, the following error shows up:

> C:\Program Files\dotnet\sdk\2.1.403\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDepedencyResolution.targets(198,5): error NETSDK1005: Assets file 'C:\...\testfx\src\Adapter\MSTest.CoreAdapter\obj\project.assets.json' doesn't have a target for '.NETFramework,Version=4.6'. Ensure that restore has run and that you have included 'net46' in the TargetFrameworks for your project. [Path to MSTest.CoreAdapter.csproj]

_and the same for PlatformServices.Interface.csproj and PlatformServices.Desktop.csproj._

Locally, I included the `'net46'` in the target frameworks for those projects (next to the `'net45'`) and it just works. I'm thinking it might be the version of the SDK I'm using for building, but not sure.

Any thought?

**Comment by mayankbansal018** [comment]
@parrainc I'm not sure what is causing this issue. I'll however will do try to take a look at it over the weekend.

**Comment by mayankbansal018** [comment]
@parrainc I just tried building this locally, & it worked fine. Were you able to build it locally or are you facing build issues locally as well?

**Comment by mayankbansal018** [comment]
@dotnet-bot test Windows / Debug Build  please

**Comment by parrainc** [comment]
@mayankbansal018 For me it builds locally in VS. In CLI throws me the error I pointed out above.

Something i'm curious is the reason why is failing here. Check the details and looks like a type exists in two assemblies that is calling it.

**Pull request closed without merging**